src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/RecommendationService.kt

```diff
@@ -230,23 +226,43 @@ internal class RecommendationService(
   }
 
   @OptIn(ExperimentalStdlibApi::class)
-  fun generatePartA(recommendationId: Long, username: String?, userEmail: String?): DocumentResponse {
-
+  suspend fun generatePartA(recommendationId: Long, username: String?, userEmail: String?): DocumentResponse {
     val recommendationEntity = updateRecommendation(null, recommendationId, username, userEmail, true)
     val userAccessResponse = recommendationEntity.data.crn?.let { userAccessValidator.checkUserAccess(it) }
     if (userAccessValidator.isUserExcludedOrRestricted(userAccessResponse)) {
       throw UserAccessException(Gson().toJson(userAccessResponse))
     } else {
-      val fileContents = templateReplacementService.generateDocFromRecommendation(recommendationEntity, DocumentType.PART_A_DOCUMENT)
+      val updatedRecommendation = updateRecommendation(recommendationEntity)
```

Comment:
Seems bit strange to be updating the case twice within the same method. Is there any way this can be refactored so there is only 1 call to update the database? Will add abit of an unnecessary overhead leaving it like this

Reply:
The first update adds the minimal details which will be required to audit that a Part A download has been attempted, AFAIK it has to come before the user access check.
The second update adds the extra fields, this has to come 'after' the user access check.

Given this, there has to be two update functions.

That said I've annotated the function with @Transactional to ensure that the framework will only make a single call and avoid overhead.